client/src/components/ProductDetails/StyleSelector.jsx

```diff
@@ -1,20 +1,45 @@
+/* eslint-disable react/prop-types */
 import React from 'react';
-import StyleThumbnail from './StyleThumbnail.jsx'
+import './prodStyles.css';
+
+
+function StyleSelector ({styles, currentStyleID, currentProductID, changeCurrentStyle}) {
+  const getCurrentProductObject = (targetId, allProducts) => {
+    return allProducts.filter((product) => {
+      return product.product_id === targetId;
+    });
+  };
+
+  const currentProductObject = getCurrentProductObject(currentProductID, styles);
```

Comment:
Are these the helper functions that you were talking about? Maybe you can help me understand how they work and is there edge case handling? For instance, what if the API returns an error or an empty array?

Reply:
These functions are temporary. Once we link the API, new helpers will be created and e can account for edge cases. Right now, I know the inputs are consistent as its linked to a file with only one object, so no need to handling edge cases yet.